monitoring/timedrift.go

```diff
@@ -133,16 +141,43 @@ func (c *timeDriftChecker) check(ctx context.Context, r health.Reporter) (err er
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
+	nodesC := make(chan serf.Member, len(nodes))
 	for _, node := range nodes {
-		drift, err := c.getTimeDrift(ctx, node)
-		if err != nil {
-			log.WithError(err).Debug("Failed to get time drift.")
-			continue
-		}
-		if isDriftHigh(drift) {
-			r.Add(c.failureProbe(node, drift))
-		}
+		nodesC <- node
+	}
+
+	var mutex sync.Mutex
+
+	var wg sync.WaitGroup
+
+	wg.Add(parallelRoutines)
+
+	for i := 0; i < parallelRoutines; i++ {
+		go func() {
+			for {
+				select {
+				case node := <-nodesC:
```

Comment:
I think it can be simpler:
```go
nodesC := make(chan serf.Member, len(nodes))
for _, node := range nodes {
	nodesC <- node
}
close(nodesC)

var wg sync.WaitGroup
var mutex sync.Mutex
wg.Add(parallelRoutines)
for i := 0; i < parallelRoutines; i++ {
	go func() {
		for node := range nodesC {
			drift, err := c.getTimeDrift(ctx, node)
			if err != nil {
				log.WithError(err).Debug("Failed to get time drift.")
				continue
			}
			if isDriftHigh(drift) {
				mutex.Lock()
				r.Add(c.failureProbe(node, drift))
				mutex.Unlock()
			}
		}
		wg.Done()
	}()
}
```